### PR TITLE
Add lazy loading example

### PR DESCRIPTION
**Pull request by leiserfg**

### CODE REVIEW

**Comment by L3MON4D3** [comment]
Nice idea, I do wonder though, does `compe` still show all snippets?

File: Examples/snippets.lua [hunk 1]
```diff
@@ -269,3 +269,24 @@ ls.snippets = {
...
+		loader.load({ include = { ft } })
+	end
+end
+
+return vim.cmd([[
+    augroup my_snips
+    autocmd!
+    augroup END
+    au my_snips BufEnter * lua _G.load_snips()
+]])
```
Comment (L3MON4D3) [inline Examples/snippets.lua:292]: Looks good, minor issues:
- I'd rather have the au inside the augroup it it's the same to you, more readable IMO.
- Is the return necessary?

Comment (leiserfg) [reply Examples/snippets.lua]: I copied it from my settings and as I use fennel the last line is always a return, and I didn't notice it :man_facepalming:

**Comment by leiserfg** [comment]
> Nice idea, I do wonder though, does `compe` still show all snippets?

It works for me, I have tested it even with several languages in the same instance.

File: Examples/snippets.lua (hunk 1 above)
Comment (L3MON4D3) [reply Examples/snippets.lua]: Ahh, okay :D

**Comment by L3MON4D3** [comment]
Very cool then, thank you!

**Pull request merged**